instrumentation/ratpack/ratpack-1.7/library/src/main/java/io/opentelemetry/instrumentation/ratpack/client/OpenTelemetryHttpClient.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.ratpack.client;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.ratpack.OpenTelemetryExecInitializer;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import ratpack.exec.Execution;
+import ratpack.http.client.HttpClient;
+import ratpack.http.client.HttpResponse;
+import ratpack.http.client.RequestSpec;
+
+final class OpenTelemetryHttpClient {
+
+  private final Instrumenter<RequestSpec, HttpResponse> instrumenter;
+
+  OpenTelemetryHttpClient(Instrumenter<RequestSpec, HttpResponse> instrumenter) {
+    this.instrumenter = instrumenter;
+  }
+
+  public HttpClient instrument(HttpClient httpClient) throws Exception {
+    return httpClient.copyWith(
+        httpClientSpec -> {
+          httpClientSpec.requestIntercept(
+              requestSpec -> {
+                Context parentOtelCtx = Context.current();
+                if (!instrumenter.shouldStart(parentOtelCtx, requestSpec)) {
+                  return;
+                }
+
+                Context otelCtx = instrumenter.start(parentOtelCtx, requestSpec);
+                Span span = Span.fromContext(otelCtx);
+                String path = requestSpec.getUri().getPath();
+                span.updateName(path);
+//                span.updateName("HTTP " + requestSpec.getMethod().getName()); // TODO use path instead of [HTTP method]
```

Comment:
Based on the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name) should be possible to leave name like `/api/users/{user_id}` instead of `HTTP GET` ?

Reply:
it depends on if path is a template (e.g. `/api/users/{user_id}`) or if it's a real url path (e.g. `/api/users/5551234`)